File: Metody obliczeniowe/L4 - Interpolacja/interpolacja_lagrange.m
function[C] = interpolacja_lagrange(X,Y)
    C = 0;
    for i=1:size(X,2)
        w=1;
        for j=1:size(X,2)
            if j~=i
                w = conv(w,[1,-X(j)]/(X(i)-X(j)));
            end
        end
        C = C + Y(i) * w;
    end
end

File: Metody obliczeniowe/L4 - Interpolacja/interpolacja_lagrange.R
function.lagrange <- function(X,Y) 
{
  C <- c(0);
  for(i in 1:length(X))
  {
    P <- c(1)
    for(j in 1:length(X))
    {
      if (j!=i)
      {
        W <- c(1,-X[j])
        P <- c(conv(P,W))/(X[i]-X[j])
      }
    }
    C <- c(C + Y[i] * P)
  }
  return (C);
}






